### PR TITLE
test: add e2e tests for rotation with azure provider

### PR DESCRIPTION
**Pull request by aramase**
**What this PR does / why we need it**:
- Adds e2e tests for auto rotation with azure provider

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: cmd/secrets-store-csi-driver/main.go [hunk 1]
```diff
@@ -74,6 +75,16 @@ func main() {
 
 	log.SetReportCaller(*logReportCaller)
 
+	// initialize metrics exporter before creating measurements
+	// Issue: https://github.com/open-telemetry/opentelemetry-go/issues/677
+	// this has been resolved in otel release v0.5.0
+	// TODO (aramase) update to latest version of otel and deps
+	m, err := metrics.NewMetricsExporter()
```
Comment (aramase) [inline cmd/secrets-store-csi-driver/main.go:82]: This limitation was identified as part of testing - https://github.com/kubernetes-sigs/secrets-store-csi-driver/pull/323. Adding this to the PR to confirm the e2e tests validate it.

**Comment by aramase** [comment]
/test pull-secrets-store-csi-driver-e2e-vault

(delay in downloading deps)

**Comment by aramase** [comment]
/hold

This PR needs an update to the bats framework as the tests for windows pass even with wrong value. With linux the test works as expected.

**Comment by aramase** [comment]
/hold cancel

seems to be an issue with my bats version 

```
➜ bats -v
Bats 1.2.1
```

/test pull-secrets-store-csi-driver-e2e-windows

File: secret1 [hunk 2]
```diff
@@ -0,0 +1 @@
+secret-8e6548fc9182
```
Comment (ritazh) [inline secret1:1]: should this be ignored?

File: secret2 [hunk 3]
```diff
@@ -0,0 +1 @@
+secret-8e6548fc9182
```
Comment (ritazh) [inline secret2:1]: same

**Pull request merged**